component/org.wso2.carbon.identity.oauth2.grant.token.exchange/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/TokenExchangeGrantHandler.java

```diff
@@ -250,7 +250,10 @@ private void validateSubjectToken(OAuthTokenReqMessageContext tokReqMsgCtx, Map<
     private String[] getScopes(JWTClaimsSet claimsSet, OAuthTokenReqMessageContext tokReqMsgCtx) {
 
         String[] requestedScopes = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getScope();
-        String[] approvedScopes = ((String) claimsSet.getClaims().get(TokenExchangeConstants.SCOPE)).split("\\s+");
+        String[] approvedScopes = new String[0];
```

Comment:
new String[0] is needed only if there is no scopes exists right? we can use Optional.

referred from AI
```java
String[] approvedScopes = Optional.ofNullable(claimsSet.getClaims().get(TokenExchangeConstants.SCOPE))
        .map(Object::toString)
        .map(scope -> scope.split("\\s+"))
        .orElse(new String[0]);
```

Reply:
Fixed with 22543c1a59d6c64534983483ff20b5c63aa59614